src/app/vertextool/qgsvertextool.cpp

```diff
@@ -313,6 +313,8 @@ QgsVertexTool::~QgsVertexTool()
   delete mVertexBand;
   delete mEdgeBand;
   delete mEndpointMarker;
+  if ( mVertexEditor )
+    delete mVertexEditor;
```

Comment:
no need to do "if (x) delete x;" - just do "delete x;"
and we probably don't need to do that anyway since the vertex editor widget has a parent (which will take care of its deletion)